gb/src/spi.rs

```diff
@@ -0,0 +1,558 @@
+                                (((pixel_b >> 5) & 0x3F) as f32 * (x_diff)*(1.0-y_diff)) + 
+                                (((pixel_c >> 5) & 0x3F) as f32 * y_diff * (1.0-x_diff)) + 
+                                (((pixel_d >> 5) & 0x3F) as f32 * x_diff * y_diff);
+                let red:f32 = (((pixel_a >> 11) & 0x1F) as f32 * (1.0-x_diff) * (1.0-y_diff)) + 
+                              (((pixel_b >> 11) & 0x1F) as f32 * (x_diff)*(1.0-y_diff)) + 
+                              (((pixel_c >> 11) & 0x1F) as f32 * y_diff * (1.0-x_diff)) + 
+                              (((pixel_d >> 11) & 0x1F) as f32 * x_diff * y_diff);
+
+                let pixel = blue as u16 | ((green as u16) << 5) | ((red as u16) << 11);
+                output_buffer[offset_counter * 2] = (pixel >> 8) as u8;
+                output_buffer[(offset_counter * 2) + 1] = (pixel & 0xFF) as u8;
+                offset_counter += 1;
+            }
+        }
+    }
+
+    fn scale_to_screen_fr(input_buffer:&[u8;SCREEN_HEIGHT*SCREEN_WIDTH * 2], output_buffer:&mut [u8;Self::TARGET_SCREEN_HEIGHT*Self::TARGET_SCREEN_WIDTH * 2]){
+        let mut buffer = input_buffer.clone();
+        let mut src_buffer = fast_image_resize::Image::from_slice_u8(
+            std::num::NonZeroU32::new(SCREEN_WIDTH as u32).unwrap(),
+            std::num::NonZeroU32::new(SCREEN_HEIGHT as u32).unwrap(),
+            &mut buffer,
+            fast_image_resize::PixelType::U16
+        ).unwrap();
+
+        let mut dst_buffer = fast_image_resize::Image::from_slice_u8(
+            std::num::NonZeroU32::new(Self::TARGET_SCREEN_WIDTH as u32).unwrap(), 
+            std::num::NonZeroU32::new(Self::TARGET_SCREEN_HEIGHT as u32).unwrap(), 
+            output_buffer, 
+            fast_image_resize::PixelType::U16
+        ).unwrap();
+
+        let mut resizer = fast_image_resize::Resizer::new(fast_image_resize::ResizeAlg::Convolution(fast_image_resize::FilterType::Bilinear));
+        resizer.resize(&src_buffer.view(), &mut dst_buffer.view_mut()).unwrap();
+    }
+
+    // implemented based on this article - https://kwojcicki.github.io/blog/NEAREST-NEIGHBOUR
+    fn scale_to_screen_nearest(input_buffer:&[u16;SCREEN_HEIGHT*SCREEN_WIDTH], output_buffer:&mut [u8;Self::TARGET_SCREEN_HEIGHT*Self::TARGET_SCREEN_WIDTH * 2]){
+        for y in 0..Self::TARGET_SCREEN_HEIGHT{
+            for x in 0..Self::TARGET_SCREEN_WIDTH{
+                let proj_x = ((1.0 / Self::SCALE) * x as f32) as usize;
+                let proj_y = ((1.0 / Self::SCALE) * y as f32) as usize;
+                unsafe{
+                    let pixel = input_buffer.get_unchecked((proj_y * SCREEN_WIDTH) + proj_x);
+                    let output_index = (y * Self::TARGET_SCREEN_WIDTH) + x;
+                    *output_buffer.get_unchecked_mut(output_index * 2) = (pixel >> 8) as u8;
+                    *output_buffer.get_unchecked_mut((output_index * 2) + 1) = (pixel & 0xFF) as u8;
+                }
+            }
+        }
+    }
```

Comment:
Add as a feature flag for the image_inter crate

Reply:
I think Ill leave this as a free function and maybe add a feature in the gb crate for fast but ugly interpolation